packages/replay/src/replay.ts

```diff
@@ -417,6 +417,15 @@ export class ReplayContainer implements ReplayContainerInterface {
     // Once this session ends, we do not want to refresh it
     if (this.session) {
       this.session.shouldRefresh = false;
+
+      // It's possible that the session lifespan is > max session lifespan
+      // because we have been buffering (which ignores expiration given that
+      // `shouldRefresh` is true). Since we flip `shouldRefresh`, the session
+      // could be considered expired due to lifespan. Update session start date
```

Comment:
contrary to the comment, session start date is _always_ current date - which should be correct? 😅